src/btree/bt_page.c

```diff
@@ -426,12 +426,39 @@ __page_reconstruct_leaf_delta(WT_SESSION_IMPL *session, WT_REF *ref, WT_ITEM *de
                 tombstone->txnid = unpack.tw.stop_txn;
                 tombstone->start_ts = unpack.tw.stop_ts;
                 tombstone->durable_ts = unpack.tw.durable_stop_ts;
-                F_SET(tombstone, WT_UPDATE_DURABLE | WT_UPDATE_RESTORED_FROM_DELTA);
+
+                if (unpack.tw.prepare) {
+                    WT_ASSERT(session,
+                      unpack.tw.prepared_id != WT_PREPARED_ID_NONE &&
+                        unpack.tw.prepare_ts != WT_TS_NONE);
+                    tombstone->prepared_id = unpack.tw.prepared_id;
+                    tombstone->prepare_ts = unpack.tw.prepare_ts;
+                    tombstone->prepare_state = WT_PREPARE_INPROGRESS;
+
+                    F_SET(tombstone, WT_UPDATE_PREPARE_RESTORED_FROM_DS);
+                    if (unpack.tw.start_txn == unpack.tw.stop_txn) {
```

Comment:
The other code has the following logic of comparing everything. I feel we need the same here as well.
```
        if ((unpack->tw.start_ts == unpack->tw.stop_ts &&
              unpack->tw.durable_start_ts == unpack->tw.durable_stop_ts &&
              unpack->tw.start_txn == unpack->tw.stop_txn)) {
 ```

Reply:
We do not have start and stop prepare_ts and prepare_id values. I feel we should continue using the `start_ts` and `stop_ts` for the time window or add `start_prepare_ts` and `stop_prepare_ts` along with the `start_prepare_id` and `stop_prepare_id`?

Reply:
as discussed offline, we'll merge this PR as is first, then split the prepare_ts and prepared_id into start_prepare_ts, stop_prepare_ts, and start_prepare_id, stop_prepare_id and change this accordingly.